File: codingjs/count7.js
function count7(n: number): number{
  if (n < 10) {
    return n === 7 ? 1 : 0;
  }
  return count7Helper(n, 0);
}
function count7Helper(dividend: number, count: number): number {
  if (dividend < 10) {
    return dividend === 7 ? count + 1 : count;
  }
  count = dividend % 10 === 7 ? count + 1 : count;
  return count7Helper(Math.floor(dividend / 10), count);
}

File: codingbat/java/count7.java
public int count7Helper(int n, int count) {
  if (n < 10) return n == 7 ? count + 1 : count;
  int remainder = n % 10;
  count += remainder == 7 ? 1 : 0;
  int quotient = n / 10;
  return count7Helper(quotient, count);
}

public int count7(int n) {
  return count7Helper(n, 0);
}

